src/d3a/d3a_core/export.py

```diff
@@ -44,9 +51,14 @@ def mkdir_from_str(directory: str, exist_ok=True, parents=True):
 
 class ExportAndPlot:
 
-    def __init__(self, root_area: Area, path: str, subdir: str):
+    def __init__(self, root_area: Area, path: str, subdir: str, endpoint_buffer):
         self.area = root_area
         self.export_data = FileExportEndpoints(root_area)
+        self.endpoint_buffer = endpoint_buffer
+        self.area_performance_index = dict()
```

Comment:
I would abstract away the KPIs from this file to their own class. The ExportAndPlot class is not supposed to do any calculation, should only write the files.